Added netstandard2.0 as target framework

ahesami opened the pull request:
- Updated project structures.
- Updated Nugets.
- Updated tests and compilation errors.

Commit:
Added netstandard2.0 as target framework

- Updated project structures and added netstandard 2.0 as target framework.
- Updated Nugets.
- Updated tests.

Commit:
Merge branch 'feature/netstandard2.0_support'

aboo committed:
Merge branch 'master' into master

aboo commented:
@ahesami please update the readme file and please check the resolved conflict

aboo commented on azure-documentdb-odata-sql/azure-documentdb-odata-sql.nuspec:
```diff
@@ -2,7 +2,7 @@
     <id>Lambda.Azure.CosmosDb.OData.Sql</id>
-    <version>2.0.4</version>
+    <version>2.0.5</version>
     <authors>Ziyou Zheng, Aboo Azarnoush</authors>
```
@ahesami add yourself as author please

aboo commented on azure-documentdb-odata-sql/azure-documentdb-odata-sql.nuspec:
```diff
@@ -22,16 +22,18 @@
-      <dependency id="Microsoft.AspNet.OData" version="6.1.0"/>
-      <dependency id="Microsoft.Azure.DocumentDB" version="1.17.0"/>
-      <dependency id="Microsoft.OData.Edm" version="7.3.1"/>
+		<group targetFramework=".NETStandard2.0">
```
@ahesami I think you should remove the group

Commit:
Updated based on review comments

- Updated read me file and added the new version.
- Updated nuspec file and removed group for netstandard2.0 target framework.
- Fixed a test after merge.
